### DISCUSSION TITLE
My obstacles and how I overcame them

### DISCUSSION THREAD

**Discussion by DiscoNova** [sbidy/wiz_light#102] [General]
I had some problems setting this up initially, and figured I might tell others "what went wrong"/"what I did wrong" and "how I got things working".

**First and foremost**; I was following the ["Install for Testing"](https://github.com/sbidy/wiz_light#install-for-testing) steps and... was somewhat perplexed by **where** exactly was the `custom_components`-folder supposed to be in. When I SSH'd to my box, I was in `/root`-folder. And this was **not** where the `custom_components`-folder was supposed to be. In fact, when I rebooted my box, any changes I had made in that folder got reset; on my setup, the folder was not "persistent". (**_edit:_** I also tried having the `custom_components`-folder in the root of my setup - i.e. `/custom_components` ... also not the correct place and not "persistent" over reboots)

After a bit of head-scratching, googling and testing, [Home Assistant documentation](https://developers.home-assistant.io/docs/creating_integration_file_structure) provided me with an answer; the `custom_components`-folder is supposed to be under the same folder that contains the `configuration.yaml`-file. For me, this meant that the *proper* location was `/config/custom_components`.

**Secondly***; I was still unable to add the devices to my `configuration.yaml`-file. Even though I followed the steps in ["Install for Testing"](https://github.com/sbidy/wiz_light#install-for-testing) ... when I tried to "Check Configuration" in Home Assistant UI, I received errors that "wiz_light" is not a recognised platform.

It turned out that I needed to add a "dummy reference" to the domain before using that domain as a `platform`. In practice, what I needed to add to my `configuration.yaml`was something not entirely unlike this:

    wiz_light:
    # Configuration for WIZ-lights
    light:
      - platform: wiz_light
        name: <NAME1>
        host: <IP1>
      - platform: wiz_light
        name: <NAME2>
        host: <IP2>

Hopefully this is of benefit for others. Naturally, your mileage may vary; e.g. after I had originally included the "dummy reference" to `wiz_light:` ... it seems like it is no longer needed; if I drop it, now the "Check Configuration" passes without problems. Personally, I've decided to keep the dummy reference in there, because it doesn't hurt my setup and currently everything works as expected :)

For me, there are still a few things I'm struggling with;
* the native iOS app for Home Assistant (both on iPhone & iPad) seems to be ignorant of both `Hold Action` and `Double Tap Action` (but they work perfectly on my computer's browser UI) and
* selecting the "effect" on computer's browser UI, the ordering of the items seems to be a mess *and* the dropdown-list provides no icons to make it easy to find what I'm looking for...

...but I *believe* these are more a matter of the Home Assistant app rather than caused by this integration.

Personally; I love this :) My Wiz-RGB bulbs were the one thing I couldn't properly interact without a number of weird IFTTT setups in the past. But now I can ... and I'm a happy camper :)

**Comment by DiscoNova** [comment]
Oh yes ... and for me, setting up the testing environment was not entirely unlike this:

    $ mkdir -p /config/custom_components
    $ mkdir -p /config/custom_components_repositories
    $ cd /config/custom_components_repositories
    $ git clone https://github.com/sbidy/wiz_light
    $ cd /config/custom_components
    $ ln -s /config/custom_components_repositories/wiz_light/custom_components/wiz_light

... which - gives me a chance of toying around with the code, updating it to the latest release, and ... always allowing me to get back to the "ground truth"-state of the repository ;)

Again - this worked for me, your mileage may vary ;)

**Comment by sbidy** [comment]
Thank you for sharing !!!
On my backlog is a Card with the title: "Create a User Documentation" 😄
